### PR TITLE
Onboard IaaS network area route

### PR DESCRIPTION
**Pull request by GokceGK**

### CODE REVIEW

File: stackit/internal/services/iaas/networkarearoute/resource.go [hunk 1]
```diff
@@ -0,0 +1,376 @@
...
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+				Validators: []validator.String{
+					validate.UUID(),
+					validate.NoSeparator(),
+				},
+			},
+			"next_hop": schema.StringAttribute{
+				Description: "IPv4 address.",
```
[the remark this answers is not in the document]
Comment (GokceGK) [reply stackit/internal/services/iaas/networkarearoute/resource.go]: I added validation for IP in next_hop and validation for CIDR in prefix

File: stackit/internal/services/iaas/networkarearoute/datasource.go [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+			},
+			"next_hop": schema.StringAttribute{
+				Description: "IPv4 address.",
+				Computed:    true,
+			},
+			"prefix": schema.StringAttribute{
+				Description: "Classless Inter-Domain Routing (CIDR).",
+				Computed:    true,
+			},
+		},
```
[the remark this answers is not in the document]
Comment (GokceGK) [reply stackit/internal/services/iaas/networkarearoute/datasource.go:130]: Updated descriptions, thx

File: stackit/internal/services/iaas/networkarearoute/resource.go [hunk 3]
```diff
@@ -0,0 +1,376 @@
...
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					validate.UUID(),
+					validate.NoSeparator(),
+				},
+			},
+			"network_area_id": schema.StringAttribute{
```
[the remark this answers is not in the document]
Comment (GokceGK) [reply stackit/internal/services/iaas/networkarearoute/resource.go:131]: Added requiresReplace.

File: stackit/internal/services/iaas/networkarearoute/resource.go [hunk 4]
```diff
@@ -0,0 +1,376 @@
...
+	var routeId string
+	var route iaas.Route
+	for _, route = range *routes.Items {
+		if !(*route.Prefix == model.Prefix.ValueString() && *route.Nexthop == model.NextHop.ValueString()) {
+			core.LogAndAddError(ctx, &resp.Diagnostics, "new static route not found in API response", fmt.Sprintf("Prefix from the route in API: %s", *route.Prefix))
+			return
+		}
+
+		routeId = *route.RouteId
+	}
```
[the remark this answers is not in the document]
Comment (GokceGK) [reply stackit/internal/services/iaas/networkarearoute/resource.go:213]: I changed the handling

File: stackit/internal/services/iaas/networkarearoute/resource.go [hunk 5]
```diff
@@ -211,21 +211,24 @@ func (r *networkAreaRouteResource) Create(ctx context.Context, req resource.Crea
...
+	if len(*routes.Items) != 1 {
+		core.LogAndAddError(ctx, &resp.Diagnostics, "Error creating network area route.", "New static route not found or more than 1 route found in API response.")
+		return
+	}
+
+	// Gets the route ID from the first element, routes.Items[0]
+	for _, route = range *routes.Items {
 		routeId = *route.RouteId
+		break
 	}
```
Comment (joaopalet) [inline stackit/internal/services/iaas/networkarearoute/resource.go]: You can get the ID at index 0 using:

```go
routeItems := *routes.Items
routeId := *routeItems[0].RouteId
```

Comment (GokceGK) [reply stackit/internal/services/iaas/networkarearoute/resource.go]: oh thx a lot <3 
I needed the route also, so changed to 

	routeItems := *routes.Items
	route := routeItems[0]
	routeId := *route.RouteId

**Review by joaopalet — APPROVED** [review]

**Pull request merged**